File: client/src/main/java/com/mthwate/conk/listener/ClearGroupListener.java
package com.mthwate.conk.listener;

import com.jme3.network.Client;
import com.mthwate.conk.WorldStore;
import com.mthwate.conk.message.ClearGroupMessage;
import com.mthwate.datlib.math.vector.Vector3i;

/**
 * @author mthwate
 */
public class ClearGroupListener extends AbstractClientListener<ClearGroupMessage> {

	@Override
	protected void onReceived(Client source, ClearGroupMessage m) {

		Vector3i start = m.getStart();
		Vector3i end = m.getEnd();

		for (int x = start.getX(); x < end.getX(); x++) {
			for (int y = start.getY(); y < end.getY(); y++) {
				for (int z = start.getZ(); z < end.getZ(); z++) {
					WorldStore.getWorld().setBlock(x, y, z, null);
				}
			}
		}

	}

}

File: client/src/main/kotlin/com/mthwate/conk/listener/ClearGroupListener.kt
package com.mthwate.conk.listener

import com.jme3.network.Client
import com.mthwate.conk.WorldStore
import com.mthwate.conk.message.ClearGroupMessage

/**
 * @author mthwate
 */
class ClearGroupListener : AbstractClientListener<ClearGroupMessage>() {

	override fun onReceived(src: Client, msg: ClearGroupMessage) {

		val start = msg.start
		val end = msg.end

		for (x in start.x..end.x - 1) {
			for (y in start.y..end.y - 1) {
				for (z in start.z..end.z - 1) {
					WorldStore.getWorld().setBlock(x, y, z, null)
				}
			}
		}

	}

}
